.test-infra/jenkins/DockerPublisher.groovy

```diff
@@ -0,0 +1,75 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import CommonJobProperties as common
+
+class DockerPublisher {
```

Comment:
Maybe we should call this class `Docker`, wdyt?

Reply:
I'm afraid it's too general. Someone would need to check the class' interface to realize what is it doing, as the class' name gives no hints.

Reply:
I think it's too general. Could you explain why `Docker` is in your opinion better?

Reply:
There is no strong opinion on my side. I think when you're using the class' interface, eg. 
`Docker.publish()` or `Docker.getFullImageName()` it's descriptive enough, whereas `DockerPublisher.publish()` is noisy (repeats "publish" word but maybe there's no need?).

On the other hand, at this point, this can become bikeshedding over a name. If you feel that `DockerPublisher` is fine, I'm fine too. This is just a nit. :)